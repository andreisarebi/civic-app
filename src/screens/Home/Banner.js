import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';
import { Entypo } from '@expo/vector-icons';
import colors from '../../styles/colors';

const Banner = props => {
  return (
    <View style={[styles.container, props.style, { backgroundColor: props.color }]}>
      <Text style={styles.title}>{props.title}</Text>
      <Text style={styles.subtitle}>{props.subtitle}</Text>
      <Entypo name={props.icon} color={colors.white} size={60} style={styles.icon} />
    </View>
  );
};

Banner.propTypes = {
  color: PropTypes.string,
  icon: PropTypes.string,
  title: PropTypes.string,
  style: PropTypes.object,
  subtitle: PropTypes.string,
  onPress: PropTypes.func,
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignSelf: 'stretch',
    minHeight: 100,
    padding: 16,
    borderRadius: 2,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
  icon: {
    opacity: 0.5,
    position: 'absolute',
    right: 20,
    bottom: 0,
  },
  title: {
    color: colors.white,
    fontSize: 24,
    fontWeight: '500',
  },
  subtitle: {
    color: colors.white,
    fontSize: 18,
    marginTop: 4,
  },
});

export default Banner;
