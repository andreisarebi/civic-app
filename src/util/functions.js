import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
