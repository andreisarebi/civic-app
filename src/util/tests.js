import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
