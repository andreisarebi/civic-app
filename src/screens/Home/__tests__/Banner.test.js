import { createSnapshot } from '../../../util/tests';
import Banner from '../Banner';

describe('Banner', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(Banner, { title: 'Register to vote!', subtitle: 'Do it!', level: 'high' });
    expect(snapshot).toMatchSnapshot();
  });
});
