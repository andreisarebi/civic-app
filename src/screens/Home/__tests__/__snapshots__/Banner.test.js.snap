// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "borderRadius": 2,
        "justifyContent": "center",
        "minHeight": 100,
        "padding": 16,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      },
      undefined,
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={undefined}
    underlayColor="black"
  >
    <Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 24,
            "fontWeight": "500",
          }
        }
      >
        Register to vote!
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 18,
            "marginTop": 4,
          }
        }
      >
        Do it!
      </Text>
      <Icon
        allowFontScaling={false}
        color="#ffffff"
        name={undefined}
        size={60}
        style={
          Object {
            "bottom": 0,
            "opacity": 0.5,
            "position": "absolute",
            "right": 20,
          }
        }
      />
    </Component>
  </TouchableHighlight>
</Component>
`;
